src/Agent.Worker/Build/TeeTFCommandManager.cs

```diff
@@ -169,16 +206,21 @@ private string FormatArguments(params string[] args)
                 }
 
                 // Add the arg.
-                arguments.Append(arg != null && arg.Contains(" ") ? $@" ""{arg}""" : $" {arg}");
+                formattedArgs.Add(arg != null && arg.Contains(" ") ? $@"""{arg}""" : $"{arg}");
             }
 
-            // Add the collection/auth parameters.
-            arguments.Append($" -collection:{Endpoint.Url.AbsoluteUri} -login:_,{accessToken} -noprompt");
+            // Add the common parameters.
+            if (!formatFlags.HasFlag(FormatFlags.OmitCollectionUrl))
```

Comment:
or you can have a overload of RunCommandAsync, one take formatflags, the other does not.